packages/next/build/swc/jest.js

```diff
@@ -56,21 +59,47 @@ const isEsmProject = packageConfig.type === 'module'
 // see https://github.com/facebook/jest/issues/9430
 const isSupportEsm = 'Module' in vm
 
-module.exports = {
+let nextConfig
+
+export default {
   process(src, filename, jestOptions) {
     if (!/\.[jt]sx?$/.test(filename)) {
       return src
     }
 
+    if (!nextConfig) {
+      loadNextConfig(jestOptions, process.cwd())
+    }
+
     let swcTransformOpts = getJestSWCOptions({
       filename,
       esm: isSupportEsm && isEsm(filename, jestOptions),
+      styledComponents: nextConfig?.experimental?.styledComponents,
     })
 
     return transformSync(src, { ...swcTransformOpts, filename })
   },
 }
 
+// exported for testing
+export function loadNextConfig(jestOptions, cwd) {
+  let transformConfig = getSwcTransformConfig(jestOptions)
+  let configPath = transformConfig?.configFile
+    ? path.posix.isAbsolute(transformConfig.configFile)
+      ? transformConfig.configFile
+      : path.join(cwd, transformConfig.configFile)
+    : findUp.sync(CONFIG_FILES, { cwd })
+  if (configPath) {
+    return require(configPath)
```

Comment:
Should we use `import()` here since `next.config.mjs` could be being used?

Reply:
It has to be synchronous 🤔

Reply:
Maybe we can use a wrapper mjs file we can execute synchronously that imports the `next.config` file and writes the needed config values to stdout which can be parsed here